subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompiler.kt

```diff
@@ -226,14 +228,27 @@ class ResidualProgramCompiler(
     }
 
     private
-    fun MethodVisitor.emitStage1Sequence(buildscript: Program.Buildscript, plugins: Program.Plugins) {
+    val Program.Stage1.fragment: ProgramSourceFragment
+        get() = when (this) {
+            is Program.Buildscript -> fragment
+            is Program.Plugins -> fragment
+            is Program.PluginManagement -> fragment
+            else -> TODO()
```

Comment:
Should this have a reason? Not clear to me if this is triggerable from user code.

Reply:
Good point @tomdcc! This will be fixed as part of https://github.com/gradle/gradle/pull/10822#pullrequestreview-296160173

Reply:
Done in https://github.com/gradle/gradle/pull/10822/commits/96d54e09a819a3041fe449ec973d290e944acf25